packages/react-components/react-tabs/src/components/Tab/useTabAnimatedIndicator.styles.ts

```diff
@@ -86,25 +86,27 @@ export const useTabAnimatedIndicatorStyles_unstable = (state: TabState): TabStat
 
   if (selected) {
     const { previousSelectedValue, selectedValue, registeredTabs } = getRegisteredTabs();
-    const previousSelectedTabRect = getRegisteredTabRect(registeredTabs, previousSelectedValue);
-    const selectedTabRect = getRegisteredTabRect(registeredTabs, selectedValue);
-
-    if (
-      selectedTabRect &&
-      previousSelectedTabRect &&
-      previousSelectedValue &&
-      lastAnimatedFrom !== previousSelectedValue
-    ) {
-      const offset = vertical
-        ? previousSelectedTabRect.y - selectedTabRect.y
-        : previousSelectedTabRect.x - selectedTabRect.x;
-
-      const scale = vertical
-        ? previousSelectedTabRect.height / selectedTabRect.height
-        : previousSelectedTabRect.width / selectedTabRect.width;
-
-      setAnimationValues({ offset, scale });
-      setLastAnimatedFrom(previousSelectedValue);
+
+    if (previousSelectedValue) {
```

Comment:
This would be a larger change and I'm not suggesting we make it right now but I think it's worth discussing as I believe it would be the most efficient implementation from a layout/style point of view.

A [ResizeObserver](https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver) will allow this code to track the sizes of all elements we care about. This will cost more memory as we'd need to maintain a `Map` of the elements and their sizes but then this code will never trigger a layout/style recalc.

Reply:
Thanks @spmonahan. Curious to learn more about it, will send you message on 1:1 chat on teams and happy to help. Can we push this PR for an fix so that we get a perf improvement on Teams client. Based on the profiling I have done, we are going to get around 5% improvement with this change so would be great if we can keep this change and work on the bigger in parallel. Let me know how you feel.

Reply:
@spmonahan The initial implementation used a ResizeObserver.  The result was significant cost registering and tracking each tab along with the overall container.  It also had problems when the Overflow/OverflowItem was involved as the making of a tab visible when selected from the overflow menu required knowing about overflow and  ignoring hidden tabs (otherwise the animation flew all over the place).  The FLIP approach used here performed well with 50+ tabs, so I don't think it is worth rewriting TabList for ResizeObserver unless someone finds a performance bar that this doesn't meet :)

I think this fix though makes sense.  When a tab is initially selected, there shouldn't be an animation anyway.  If this isn't pressing, I will pull and review this when I'm back from vacation.

Reply:
@spmonahan updated the PR with yarn change and moved `lastAnimatedFrom !== previousSelectedValue` check along with `previousSelectedValue` as well. Appreciate all the support.